### PR TITLE
[doc] Improved readme and all contributors

### PR DESCRIPTION
**Pull request by adangel**
## Describe the PR

I tried to improve our README. And also add all-contributors (https://allcontributors.org/) (but without the bot yet).

Resources used:
* https://www.freecodecamp.org/news/how-to-write-a-good-readme-file/
* https://www.makeareadme.com/
* https://allcontributors.org/

Preview: https://github.com/adangel/pmd/blob/improved-readme/README.md
Preview credits: https://github.com/adangel/pmd/blob/improved-readme/docs/pages/pmd/projectdocs/credits.md

Much more colorful :smile: 

Questions, suggestions:
* Should we add the pmd logo at the top?
* Should we really use all contributors?
  * It's easy to find the code contributors, but much harder to find other contributors (issue reporter, tester, ...)
  * add another language/framework to the mix: nodejs...
* The Demo looks very unspectacular - I guess, we should create a (gif) video
  * https://github.com/icholy/ttygif
  * https://asciinema.org/
  * https://terminalizer.com/ (with this fix: https://github.com/faressoft/terminalizer/pull/154)
  * https://github.com/phw/peek
  * https://github.com/HasinduLanka/screencorder
  * https://github.com/xlmnxp/blue-recorder 
* Not sure, if we should keep the snapshot of pmd-eclipse-plugin here...

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by adangel** [comment]
I finally found a program, that kind of works to record screens/single windows under wayland: https://github.com/HasinduLanka/screencorder

Here's my try:

pmd-demo-5fps

File: .all-contributorsrc [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      "contributions": [
+        "code",
+        "doc"
+      ]
+    },
+    {
+      "login": "oowekyala",
+      "name": "Clément Fournier",
+      "avatar_url": "https://avatars.githubusercontent.com/u/24524930?v=4",
+      "profile": "https://www.linkedin.com/in/le-clement-fournier/",
```
Comment (oowekyala) [inline .all-contributorsrc:27]: I'd rather my github profile be listed here. Is it possible to edit this file manually?

File: README.md [hunk 2]
```diff
@@ -7,18 +7,73 @@
...
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/a674ee8642ed44c6ba7633626ee95967)](https://www.codacy.com/app/pmd/pmd?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=pmd/pmd&amp;utm_campaign=Badge_Grade)
 [![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v2.0%20adopted-ff69b4.svg)](code_of_conduct.md) 
-
-## About
+[![Documentation (latest)](https://img.shields.io/badge/docs-latest-green)](https://pmd.github.io/latest/)
 
 **PMD** is a source code analyzer. It finds common programming flaws like unused variables, empty catch blocks,
-unnecessary object creation, and so forth. It supports Java, JavaScript, Salesforce.com Apex and Visualforce,
+unnecessary object creation, and so forth. It supports many languages. It can be extended with custom rules.
+It uses JavaCC and Antlr for parsing source into AST and runs rules against it to find violations.
```
Comment (oowekyala) [inline README.md:14]:
```suggestion
It uses JavaCC and Antlr to parse source files into abstract syntax trees (AST) and runs rules against them to find violations.
```

**Review by oowekyala — COMMENTED** [review]
Thanks, this looks great :)

File: .all-contributorsrc (hunk 1 above)
Comment (adangel) [reply .all-contributorsrc]: I've changed it

**Comment by adangel** [comment]
I've now added the first 100 github contributors with the following script (maybe it's useful for others):

```
export PRIVATE_TOKEN=....
curl https://api.github.com/repos/pmd/pmd/contributors?per_page=100 > pmd-contributors.json
for i in $(cat pmd-contributors.json | jq -r '.[].login'); do
  echo $i
  npx all-contributors add $i code
  sleep 1
done
```

This API only provides the first top 100 contributors (see also https://github.com/pmd/pmd/graphs/contributors)

**Comment by adangel** [comment]
Better screencast:

https://user-images.githubusercontent.com/1573684/131829918-3299e692-51f1-4396-aab1-2dba1d508987.mp4

**Comment by adangel** [comment]
It seems, that only gifs work in README.md files, although you can upload videos in comments (https://github.blog/2021-05-13-video-uploads-available-github/)

For creating the screencast, I used gnome and https://github.com/EasyScreenCast/EasyScreenCast
This will create a webm video.

With ffmpeg, it can be converted into mp4: `ffmpeg -i pmd-demo.webm -r 15 -s 708x484 pmd-demo.mp4`

Or into a high quality gif (http://blog.pkh.me/p/21-high-quality-gif-with-ffmpeg.html):

```
ffmpeg -i pmd-demo.webm -vf "fps=3,palettegen" -y pmd-demo-palette.png
ffmpeg -i pmd-demo.webm -i pmd-demo-palette.png -lavfi "fps=3 [x]; [x][1:v] paletteuse" -y pmd-demo.gif
```

**Comment by adangel** [comment]
> Should we add the pmd logo at the top?

We should probably do this for pmd7, but not for pmd6

> Should we really use all contributors?
>    It's easy to find the code contributors, but much harder to find other contributors (issue reporter, tester, ...)
>    add another language/framework to the mix: nodejs...

I guess, it makes sense to add the bot (https://allcontributors.org/docs/en/bot/overview) in order to easily add new contributors.

We have however older contributors (https://github.com/pmd/pmd/blob/master/docs/pages/pmd/projectdocs/credits.md) which we need to migrate....

If the list is too big in README.md, we might move it into CONTRIBUTORS.md


> The Demo looks very unspectacular - I guess, we should create a (gif) video

The new demo is simple and basic... at least a moving picture

> Not sure, if we should keep the snapshot of pmd-eclipse-plugin here...

Removed it.

**Comment by adangel** [comment]
Just noticed: The readme in the pmd7 branch already has the new logo : https://github.com/pmd/pmd/blob/pmd/7.0.x/README.md

**Comment by adangel** [comment]
Here are the scriptlets I used to get the "issue"-contributors from github:

```shell
# download the issues... page 1 to 35, 100 per page
for page in $(seq 1 35); do
    curl \
      -H "Accept: application/vnd.github.v3+json" \
      "https://api.github.com/repos/pmd/pmd/issues?per_page=100&page=${page}&direction=asc&state=all" > pmd-issues-${page}.json
done

# extract the users
for page in $(seq 1 35); do
   jq -r '.[].user.login' pmd-issues-${page}.json >> pmd-issues-users.txt
done

# remove duplicates
sort pmd-issues-users.txt | uniq > pmd-issues-users-uniq.txt

# collect already added contributors
jq -r '.contributors[].login' .all-contributorsrc > contributors-users.txt
sort contributors-users.txt | uniq > contributors-users-uniq.txt

# contributors, that are already in .all-contributorsrc but additionally created bugs
comm -1 -2 contributors-users-uniq.txt pmd-issues-users-uniq.txt

# new contributors, who only created bugs
comm -1 -3 contributors-users-uniq.txt pmd-issues-users-uniq.txt

npm i
export PRIVATE_TOKEN=...

# Add new "bug" contributors:
for i in $(comm -1 -3 contributors-users-uniq.txt pmd-issues-users-uniq.txt); do
    echo $i;
    npx all-contributors add $i bug
    sleep 0.5
done
```

**Pull request merged**